lib/ui/channel_buffers.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of dart.ui;
+
+/// A saved platform message for a channel with its callback.
+class StoredMessage {
+  final ByteData _data;
+  final PlatformMessageResponseCallback _callback;
+
+  StoredMessage(this._data, this._callback);
+  ByteData get data => _data;
+  PlatformMessageResponseCallback get callback => _callback;
+}
+
+/// A fixed-size circular queue.
+class RingBuffer<T> {
```

Comment:
This should be private (`_RingBuffer`) - it will still be visible to everything else in `dart:ui`, but we don't want to expose this externally.

It'd still be good to write docs for all the methods in here, but it's not as critical that it be high quality user facing documentation if it's private.

Reply:
Done.